### PR TITLE
chore(weave): Enable local storage cache for nodes

### PR DESCRIPTION
**Pull request by tssweeney**
This PR enables the basic client to use local storage as a persistent cache - making reloading common nodes across refreshes nearly instant

### CODE REVIEW

File: weave-js/src/core/cache/localStorageBackedLRU.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        }
+
+        return setDone;
+
+    }
+    
+    private removeLeastRecentlyUsed(): void {
+        const key = localStorage.key(0)
+        if (key) {
+            console.log("Evicting key", key)
```
Comment (dannygoldstein) [inline weave-js/src/core/cache/localStorageBackedLRU.ts:34]: dev tracks

File: weave-js/src/core/cache/localStorageBackedLRU.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                    break; // Chrome
+                case 1014:
+                    // Firefox
+                    if (e.name === 'NS_ERROR_DOM_QUOTA_REACHED') {
+                        quotaExceeded = true;
+                    }
+                    break;
+            }
+        } else if (e.number === -2147024882) {
+            // Internet Explorer 8
```
Comment (dannygoldstein) [inline weave-js/src/core/cache/localStorageBackedLRU.ts:82]: do we need to add safari?

**Review by dannygoldstein — APPROVED** [review]

File: weave-js/src/core/cache/localStorageBackedLRU.ts (hunk 2 above)
Comment (tssweeney) [reply weave-js/src/core/cache/localStorageBackedLRU.ts:82]: added comment - same as chrome

File: weave-js/src/core/cache/localStorageBackedLRU.ts (hunk 1 above)
Comment (tssweeney) [reply weave-js/src/core/cache/localStorageBackedLRU.ts]: fixed

File: weave-js/src/core/client/basic.ts [hunk 3]
```diff
@@ -244,6 +254,10 @@ export class BasicClient implements Client {
       ) => {
         observable.hasResult = true;
         observable.lastResult = result;
+        if (result != null) {
+          // Skip caching null results for now.
```
Comment (tssweeney) [inline weave-js/src/core/client/basic.ts:258]: Can re-enable this in theory if the other fixes work

**Pull request merged**